services/git-service-impl/src/main/java/io/fabric8/launcher/service/gitlab/impl/GitLabServiceFactoryImpl.java

```diff
@@ -0,0 +1,36 @@
+package io.fabric8.launcher.service.gitlab.impl;
+
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.fabric8.launcher.base.EnvironmentSupport;
+import io.fabric8.launcher.base.identity.Identity;
+import io.fabric8.launcher.base.identity.IdentityFactory;
+import io.fabric8.launcher.base.identity.TokenIdentity;
+import io.fabric8.launcher.service.gitlab.api.GitLabService;
+import io.fabric8.launcher.service.gitlab.api.GitLabServiceFactory;
+
+import static io.fabric8.launcher.service.gitlab.api.GitLabEnvVarSysPropNames.GITLAB_PRIVATE_TOKEN;
+
+/**
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+@ApplicationScoped
+public class GitLabServiceFactoryImpl implements GitLabServiceFactory {
+
+    @Override
+    public GitLabService create(Identity identity) {
+        if (!(identity instanceof TokenIdentity)) {
+            throw new IllegalArgumentException("GitLabService supports only TokenIdentity. Not supported:" + identity);
+        }
+        return new GitLabServiceImpl((TokenIdentity) identity);
+    }
+
+    @Override
+    public Optional<Identity> getDefaultIdentity() {
+        // Try using the provided Github token
+        String token = EnvironmentSupport.INSTANCE.getEnvVarOrSysProp(GITLAB_PRIVATE_TOKEN);
+        return token == null ? Optional.empty() : Optional.of(IdentityFactory.createFromToken("Private-Token", token));
```

Comment:
sth like `return Optional.ofNullable(token).map(token -> IdentityFactory.createFromToken("Private-Token", token))` ? but that's more a matter of taste I guess.